hphp/runtime/vm/jit/irlower-conv.cpp

```diff
@@ -198,17 +198,23 @@ void cgConvDblToInt(IRLS& env, const IRInstruction* inst) {
   v << cvttsd2siq{src, d};
   v << cmpq{indef, d, sf};
 
+  // For ARM, use the cc which ignores unordered results for the first
+  // condition. Then use the ARM equivalent cc to check for the unordered
+  // result in the second condition.
+  ConditionCode cc_nb = (arch() == Arch::ARM) ? CC_GE : CC_NB;
+  ConditionCode cc_p = (arch() == Arch::ARM) ? CC_O : CC_P;
```

Comment:
I'd say that this new fix boils down to the inherent differences with ARM. Firstly, there is not parity bit. Instead if providing an explicit flag and then simple condition codes to test that flag, ARM provides multiple sets of condition codes that allow you to test >= without concern for unordered results and then with concern for unordered results.

Secondly, many of the ARM condition codes have multiple meanings depending on whether the context for the comparison was int or fp.

The table I added to abi-arm.h is a great reference for the ARM cc's, their meanings, and how we've mapped them to the JIT cc's and the JIT cc meanings.

With that background out of the way, the first conditional assignment of  `cc_nb` above sets it to `CC_GE` for ARM, because that maps to the ARM cc (ge) which doesn't take into account unordered results. This code here in irlower-conv.cpp doesn't consider unordered results in the first test using `cc_nb`. The second conditional assignment of `cc_p` above sets it to `CC_O` for ARM, because that it is mapped to `vs`, which is normally the overflow test in the integer context, but for the fp context it indicates an unordered result.

I tried to work this out in the actual mapping from ARM to JIT cc's, but the dual meaning of ARM cc's made it tough. I also considered introducing a new group of JIT cc's that would allow us to capture the semantics of ARM's different cc's that are concerned with unordered results too, but I thought this was way to heavy-handed.

Anyhow, my understanding of x86_64 cc's is shaky at best, so if indeed X64 is more similar to ARM than I think, then perhaps this can actually be solved in a more portable way through mapping and/or adding new JIT cc's.

And to answer your question, I wouldn't call it a coincidence, per se, but it sure is funny how such a simple change fixed the problem. Switching on Arch::ARM in the location which is producing the checks is a subtle way to take into account the context of operations being performed at that location.